### PR TITLE
feat(transformer/typescript): support for transform enum

### PR DESCRIPTION
**Pull request by Dunqing**
The current implementation is copied from the previous implementation

### CODE REVIEW

**Comment by Dunqing** [comment]
* **#2997** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2997?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_transformer/src/typescript/annotations.rs [hunk 1]
```diff
@@ -325,4 +327,236 @@ impl<'a> TypeScriptAnnotations<'a> {
...
+    ///   X
+    /// }
+    /// ```
+    /// ```JavaScript
+    /// var Foo = ((Foo) => {
+    ///   const X = 0; Foo[Foo["X"] = X] = "X";
+    ///   return Foo;
+    /// })(Foo || {});
+    /// ```
+    fn transform_ts_enum(
```
Comment (Boshen) [inline crates/oxc_transformer/src/typescript/annotations.rs:342]: Can we isolate this in a new file?

File: crates/oxc_transformer/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    "import",
+    "null",
+    "true",
+    "false",
+    "in",
+    "instanceof",
+    "typeof",
+    "void",
+    "delete",
+];
```
Comment (Boshen) [inline crates/oxc_transformer/src/utils.rs:51]: These should be in `oxc_syntax`. crates/oxc_syntax/src/keyword.rs

File: crates/oxc_transformer/src/typescript/annotations.rs (hunk 1 above)
Comment (milesj) [reply crates/oxc_transformer/src/typescript/annotations.rs:342]: Yeah I was gonna do this in an enums file.

Comment (Dunqing) [reply crates/oxc_transformer/src/typescript/annotations.rs]: Yes, i moved this to `enum.rs`

**Review by Boshen — APPROVED** [review]

**Pull request merged**